GVFS/GVFS.Common/Prefetch/Git/DiffHelper.cs

```diff
@@ -369,9 +369,9 @@ private void EnqueueFileAddOperation(ITracer activity, DiffTreeResult operation)
             // Each filepath should be case-insensitive unique. If there are duplicates, only the last parsed one should remain.
             if (!this.filesAdded.Add(operation.TargetPath))
             {
-                foreach (KeyValuePair<string, HashSet<string>> kvp in this.FileAddOperations)
+                foreach (KeyValuePair<string, HashSet<PathWithMode>> kvp in this.FileAddOperations)
                 {
-                    if (kvp.Value.Remove(operation.TargetPath))
+                    if (kvp.Value.Any(m => m.Path == operation.TargetPath))

```

Comment:
This line used to remove the entry from this.FileAddOperations, and now it's just checking for existence (and `Any` will be linear time rather than constant).

Can you switch back to `Remove` and create a temporary `PathWithMode` with a dummy mode?  With the change to use the path for GetHashCode\Equals you should still be able to call `Remove`.